### PR TITLE
Gas estimates

### PR DESCRIPTION
**Pull request by 0xngmi**

### CODE REVIEW

File: src/components/Aggregator/adapters/kyberswap.ts [hunk 1]
```diff
@@ -77,3 +77,8 @@ export async function swap({ signer, from, rawQuote, chain }) {
 	return tx;
 }
 export const getTxData = ({ rawQuote }) => rawQuote?.encodedSwapData;
+
+export const getTx = ({ rawQuote }) => ({
+	to: rawQuote.routerAddress,
+	data: rawQuote.encodedSwapData
```
Comment (0xngmi) [inline src/components/Aggregator/adapters/kyberswap.ts:83]: value missing?

File: src/components/Aggregator/hooks/useEstimateGas.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+						to: txData.to,
+						data: txData.data,
+						...(isNative ? { value: '0x' + BigNumber(amount).toString(16) } : {})
+					},
+					['trace', 'vmTrace']
+				]),
+				'latest'
+			];
+			const res = await provider.send('trace_callMany', callParams);
+			const swapTx = last<{ trace: Array<{ result: { gasUsed: string }; error: string }> }>(res);
```
Comment (0xngmi) [inline src/components/Aggregator/hooks/useEstimateGas.ts:44]: does last just pick last element in array?

File: src/components/Aggregator/index.tsx [hunk 3]
```diff
@@ -638,6 +649,7 @@ export function AggregatorContainer({ tokenlist }) {
 	let normalizedRoutes = [...(routes || [])]
 		?.map(fillRoute)
 		.filter(({ fromAmount, amount: toAmount }) => Number(toAmount) && amountWithDecimals === fromAmount)
```
Comment (0xngmi) [inline src/components/Aggregator/index.tsx:651]: better to join the 2 filter()

Comment (0xngmi) [reply src/components/Aggregator/index.tsx:651]: more efficient

File: src/components/WalletProvider/index.tsx [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
 import styled from 'styled-components';
 import { allChains } from './chains';
 
 const { provider, chains } = configureChains(
 	[
+		chain.arbitrum,
+		{
+			...chain.mainnet,
+			rpcUrls: {
+				default: 'https://eth.llamarpc.com'
```
Comment (0xngmi) [inline src/components/WalletProvider/index.tsx:15]: where are these rpcs used?

**Comment by 0xngmi** [comment]
went testing it ran into some weird flickering in the route selection

File: src/components/Aggregator/hooks/useEstimateGas.ts (hunk 2 above)
Comment (vrtnd) [reply src/components/Aggregator/hooks/useEstimateGas.ts:44]: yes, easily to read than arr[arr.length - 1]

File: src/components/WalletProvider/index.tsx (hunk 4 above)
Comment (vrtnd) [reply src/components/WalletProvider/index.tsx:15]: for wagmi hooks

File: src/components/Aggregator/adapters/kyberswap.ts (hunk 1 above)
Comment (vrtnd) [reply src/components/Aggregator/adapters/kyberswap.ts:83]: i use amountIn as value in estimate hook

File: src/components/WalletProvider/index.tsx (hunk 4 above)
Comment (0xngmi) [reply src/components/WalletProvider/index.tsx:15]: are they used for estimateGas()?

Comment (vrtnd) [reply src/components/WalletProvider/index.tsx:15]: yes, the used for estimateGas inside sendTx

**Review by vrtnd — APPROVED** [review]

**Pull request merged**